nutanix/data_source_nutanix_category_key.go

```diff
@@ -0,0 +1,81 @@
+package nutanix
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	v3 "github.com/terraform-providers/terraform-provider-nutanix/client/v3"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+func dataSourceNutanixCategoryKey() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNutanixCategoryKeyRead,
+
+		Schema: map[string]*schema.Schema{
+			"system_defined": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"api_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"values": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceNutanixCategoryKeyRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Reading CategoryKey: %s", d.Get("name").(string))
+
+	// Get client connection
+	conn := meta.(*Client).API
+
+	// Make request to the API
+	resp, err := conn.V3.GetCategoryKey(d.Get("name").(string))
+
+	if err != nil {
+		if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
+			d.SetId("")
+		}
+		return err
+	}
+
+	d.Set("api_version", utils.StringValue(resp.APIVersion))
+	d.Set("name", utils.StringValue(resp.Name))
+	d.Set("description", utils.StringValue(resp.Description))
+	d.Set("system_defined", utils.BoolValue(resp.SystemDefined))
+
+	d.SetId(utils.StringValue(resp.Name))
+
+	list, err := conn.V3.ListCategoryValues(d.Get("name").(string), &v3.CategoryListMetadata{
+		Kind:   utils.StringPtr("category"),
+		Length: utils.Int64Ptr(500),
+	})
```

Comment:
Also, this is kinda hard to read, I know it seems easier to send the struct as an argument instead of creating a variable, but it gives unnecessary length to the call